docs/rust-client/custom_note_how_to.md

```diff
@@ -0,0 +1,443 @@
+# How To Create a Custom Note on Miden
+
+*Creating custom notes*
+
+## Overview
+
+In this guide, we will create a custom note on Miden that can only be consumed by someone who knows the preimage of the hash stored in the note. This approach securely embeds assets into the note and restricts spending to those who possess the correct secret number.
+
+By following the steps below and using the Miden Assembly code and Rust example, you will learn how to:
+
+* Mint and store assets within a custom note.
+* Require a secret preimage to consume the note.
+* Leverage Miden’s privacy features to keep transaction details private.
+
+Unlike Ethereum, where transaction details are publicly visible in the mempool, Miden enables you to partially or completely hide transaction details.
+
+
+## What we'll cover
+
+- Creating custom note logic
+- Consuming notes
+
+## Step-by-Step Process
+
+### 1. Creating Two Accounts: Alice & Bob
+
+First, we create two basic accounts for the two users:
+- **Alice:** The account that creates and funds the custom note.
+- **Bob:** The account that will consume the note if he knows the secret.
+
+### 2. Deploying a Fungible Faucet
+
+Next, we deploy a fungible faucet. This faucet is used to mint assets (tokens) that will be included in the note. The asset is identified by a token symbol (e.g., MID) and minted to Alice’s account. This ensures that Alice has the necessary tokens to create the note.
+
+### 3. Hashing the Secret Number
+
+The security of the custom note hinges on a secret number. Here, we will:
+- Choose a secret number (for example, an array of four field elements).
```

Comment:
Can you just say integers instead of `field elements`?

I would try to abstract all zk-jargon away